### PR TITLE
Digikam +friends version bump to 4.14.0

### PR DESCRIPTION
**Pull request by a17r**
Taken from kde overlay (as a note to maintainers to drop mask and ebuilds as soon as PR is merged):
- kde-apps/libkface-15.08.2 (-r1 adding opencv3 patch)
- kde-apps/libkgeomap-15.08.2

I didn't bump the respective duplicates *media-libs*/libkface and *media-libs*/libkgeomap anymore.

Fixes various bugs (most about opencv3 support), should get some runtime testing though:
https://bugs.gentoo.org/show_bug.cgi?id=554490
https://bugs.gentoo.org/show_bug.cgi?id=554494
https://bugs.gentoo.org/show_bug.cgi?id=554496
https://bugs.gentoo.org/show_bug.cgi?id=559762
https://bugs.gentoo.org/show_bug.cgi?id=557174

### CODE REVIEW

**Comment by mgorny** [comment]
@gentoo/kde

File: media-gfx/digikam/digikam-4.14.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+	mv "${WORKDIR}/${MY_P}/doc/${PN}/digikam" doc-digikam/en || die
+	echo "add_subdirectory( en )" > doc-showfoto/CMakeLists.txt || die
+	mv "${WORKDIR}/${MY_P}/doc/${PN}/showfoto" doc-showfoto/en || die
+	sed -i -e 's:../digikam/:../../doc-digikam/en/:g' doc-showfoto/en/index.docbook || die
+
+	# prepare the translations
+	mv "${WORKDIR}/${MY_P}/po" po || die
+	find po -name "*.po" -and -not -name "digikam.po" -delete || die
+
+	echo "find_package(Msgfmt REQUIRED)" >> CMakeLists.txt || die
```
Comment (mgorny) [inline media-gfx/digikam/digikam-4.14.0.ebuild]: Suggest heredocs here, harder to typo.

File: media-gfx/digikam/digikam-4.14.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+}
+
+src_configure() {
+	# LQR = only allows to choose between bundled/external
+	local mycmakeargs=(
+		-DENABLE_LCMS2=ON
+		-DENABLE_OPENCV3=ON
+		-DWITH_LQR=ON
+		-DWITH_LENSFUN=ON
+		$(cmake-utils_use_enable addressbook KDEPIMLIBSSUPPORT)
```
Comment (mgorny) [inline media-gfx/digikam/digikam-4.14.0.ebuild]: usex love? ;-)

File: media-gfx/digikam/digikam-4.14.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+	kde4-base_src_compile ${mytargets}
+}
+
+src_install() {
+	kde4-base_src_install
+
+	if use doc; then
+		# install the api documentation
+		insinto /usr/share/doc/${PF}/
+		doins -r ${CMAKE_BUILD_DIR}/api/html
```
Comment (mgorny) [inline media-gfx/digikam/digikam-4.14.0.ebuild]: dodoc.

File: media-gfx/digikam/digikam-4.14.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+	fi
+}
+
+pkg_postinst() {
+	kde4-base_pkg_postinst
+
+	if use doc; then
+		einfo "The digikam api documentation has been installed at /usr/share/doc/${PF}/html"
```
Comment (mgorny) [inline media-gfx/digikam/digikam-4.14.0.ebuild]: isn't that kinda... obvious?

**Comment by devurandom** [comment]
> Taken from kde overlay

Are you sure these ebuilds are already in the KDE overlay? I don't see them there [^1], and hence cannot easily test right now.

[^1]: https://github.com/gentoo-mirror/kde/tree/master/kde-apps/libkface

**Comment by a17r** [comment]
@devurandom 15.08.2 is currently masked in kde overlay. I took libkface from overlay but needed to bump it here to include opencv3 support.

**Comment by devurandom** [comment]
So kde-apps/libkface-15.08.2::kde == kde-apps/libkface-15.08.2-r1::gentoo?

Next issue: The KDE overlay seems to be missing [^1] the media-gfx/digikam-4.14.0 ebuilds.

[^1]: https://github.com/gentoo-mirror/kde/tree/master/media-gfx

**Comment by a17r** [comment]
No, you take everything from here. ;)

**Comment by devurandom** [comment]
So it is in fact not taken from the KDE overlay at all? In this case: Could you please add it to the overlay first, to make testing for me easier? I am especially interested in confirming that b.g.o#554490 is fixed by these changes.

**Comment by a17r** [comment]
I can't commit to overlay. ;) As I said, you can take any commit from here as a patch like so: https://github.com/a17r/gentoo/commit/845f7998e0ae306a44d91324b59b3e26115db933.patch (this concerns bug 554490)

File: media-gfx/digikam/digikam-4.14.0.ebuild (hunk 1 above)
Comment (a17r) [inline media-gfx/digikam/digikam-4.14.0.ebuild]: This is ultimately the last release of kde4-based digikam. It remains to be seen how digikam-5.0.0 will organise docs once it is released.

**Comment by mgorny** [comment]
@kensington, want me to give @a17r commit access to the overlay?

**Comment by a17r** [comment]
libkface is currently very busy detecting faces on the family PC - seems to work. But I'm a first-time user of that feature.

File: kde-apps/libkface/libkface-15.08.2-r1.ebuild [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
+
+inherit kde4-base
+
+DESCRIPTION="Qt/C++ wrapper around LibFace to perform face recognition and detection"
+HOMEPAGE="http://api.kde.org/4.x-api/kdegraphics-apidocs/libs/libkface/libkface/html/index.html"
```
Comment (hasufell) [inline kde-apps/libkface/libkface-15.08.2-r1.ebuild]: homepage dead?

Comment (a17r) [inline kde-apps/libkface/libkface-15.08.2-r1.ebuild]: That must be new, maybe temporary issue because of master branch moving to frameworks

**Comment by Beep6581** [comment]
Hello

I tried compiling the ebuild from here, but it fails with "add_custom_target cannot create target "pofiles" because another target with the same name already exists.  The existing target is a custom target created in source directory". This is the same error I get when trying to compile the 4.12.0 ebuild in portage renamed to 4.14.0.

emerge --info https://bpaste.net/show/e176455280c4
complete build log http://filebin.net/b5y0w7aa3x/build.log.txt

**Comment by a17r** [comment]
@Beep6581, I don't think that issue can be solved in digikam. What version of kdelibs is installed? Did you try with kdelibs-4.14.13-r1 already?

**Comment by Beep6581** [comment]
Thank you, upgrading to kdelibs-4.14.13-r1 let me emerge digikam-4.14.0 successfully.

**Comment by kensington** [comment]
Thanks so much for you work! I went ahead and did the library pkgmove so adjusted for that, and restored the subslot where it was removed (to avoid triggering an unnecessary rebuild).

**Pull request closed without merging**